Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -142,28 +158,45 @@ def set_up(self, initial, f, g, step_size, **kwargs):
         self.g = g
 
         if isinstance(f, ZeroFunction) and isinstance(g, ZeroFunction):
-            raise ValueError('You set both f and g to be the ZeroFunction and thus the iterative method will not update and will remain fixed at the initial value.')
+            raise ValueError(
+                'You set both f and g to be the ZeroFunction and thus the iterative method will not update and will remain fixed at the initial value.')
 
         # set step_size
-        self.set_step_size(step_size=step_size)
+        if step_size is None:
+            self.step_size_rule = ConstantStepSize(
+                self._calculate_default_step_size())
+        elif isinstance(step_size, Real):
+            self.step_size_rule = ConstantStepSize(step_size)
+        elif isinstance(step_size, StepSizeRule):
+            self.step_size_rule = step_size
+        else:
+            raise TypeError(
+                '`step_size` must be `None`, a real float or a child class of :meth:`cil.optimisation.utilities.StepSizeRule`')
```

Comment:
I would leave this out. We can describe it in the docstring and probably use type hinting. 
It'll later fail if the object passed doesn't have `get_step_size`